modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/freelist/PagesList.java

```diff
@@ -2033,6 +2033,9 @@ public synchronized boolean add(long pageId) {
             int stripeIdx = (int)pageId & (STRIPES_COUNT - 1);
 
             synchronized (stripeLocks[stripeIdx]) {
+                if (size >= MAX_SIZE)
```

Comment:
It's a redundant check. There already exists a stripe size check under the lock.

Reply:
not this, but upper, if so - i agree.